src/Mantasflowers.WebApi/Controllers/UserController.cs

```diff
@@ -1,206 +1,98 @@
-            };
+            var response = await _userService.GetDetailedUserByUidAsync(uid, emailData.Email, emailData.IsEmailVerified);
 
-            return HandleException(async () =>
-            {
-                await _fbService.SetCustomUserClaimsAsync(request.Uid, claims);
-                return Ok(ResponseMsg.CustomClaimSet);
-            });
-        }
-
-        [HttpPost("create-user")]
-        public IActionResult CreateUser([FromBody] PostCredentialsRequest request)
-        {
-            return HandleException(async () =>
-            {
-                var user = await _fbService.CreateUserAsync(request.Email, request.Password);
-                return Ok(user);
-            });
+            return Ok(response);
         }
 
         /// <summary>
-        /// 
+        /// [NOT IMPLEMENTED] Completely delete user and all its information.
         /// </summary>
-        /// <param name="request">For testing purposes Uid: G46RSuLIvbUjaWJyokcCMDSCeVj1</param>
-        /// <returns></returns>
-        [Authorize(Roles = "admin")]
-        [HttpPost("user/by-uid")]
-        public IActionResult GetUserByUid([FromBody] PostUidRequest request)
+        [HttpPost("delete")]
+        public IActionResult DeleteUser()
         {
-            return HandleException(async () =>
-            {
-                var user = await _fbService.GetUserByUidAsync(request.Uid);
-                return Ok(user);
-            });
+            // TODO: this needs to delete user and any data related to it in our DB + firebaseUser
+            throw new NotImplementedException("NOT IMPLEMENTED");
         }
 
-        /// <summary>
-        /// 
-        /// </summary>
-        /// <param name="email">For testing purposes: johnny@john.jo</param>
-        /// <returns></returns>
-        [Authorize(Roles = "admin")]
-        [HttpGet("user/by-email")]
-        public IActionResult GetUserByEmail(string email)
+        // TODO: transactions?
+        [AllowAnonymous]
+        [HttpPost("create")]
+        [ProducesResponseType(typeof(PostCreateUserResponse), StatusCodes.Status201Created)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> CreateUser(PostCreateUserRequest postCreateUserRequest)
         {
-            return HandleException(async () =>
-            {
-                var user = await _fbService.GetUserByEmailAsync(email);
-                return Ok(user);
-            });
-        }
+            UserRecord firebaseUser = null;
+            firebaseUser = await _fbService.CreateUserAsync(postCreateUserRequest.Email, postCreateUserRequest.Password);
 
-        [Authorize]
-        [HttpPost("update-user/email")]
-        public IActionResult UpdateUserEmail([FromBody] PostUpdateEmailRequest request)
-        {
-            return HandleException(async () =>
+            if (string.IsNullOrWhiteSpace(firebaseUser.Uid))
```

Comment:
In a way, an unreachable code, since Firebase would throw an exception if something went wrong (be it the case with Uid). Else, there is no need to check for this.

Reply:
I think github is confused, this is how updateUserEmail controller method looks now:
```csharp
[Authorize]
[HttpPost("update/email")]
[ProducesResponseType(typeof(UpdateEmailResponse), StatusCodes.Status200OK)]
public async Task<IActionResult> UpdateUserEmail(PostUpdateEmailRequest request)
{
    string uid = User.GetUidFromJwt();

    var user = await _fbService.UpdateUserEmailAsync(uid, request.Email);
    var resposne = new UpdateEmailResponse()
    {
        Email = user.Email
    };
    
    return Ok(resposne);
}
```